### PR TITLE
Remove Travis integration and add PyPI release action

### PR DESCRIPTION
**Pull request by punchcutter**
This should now trigger a PyPI release when a new release tag is pushed.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+      contents: write
+
+    steps:
+      - name: Retrieve release distributions
+        uses: actions/download-artifact@v4
+        with:
+          name: release-dists
+          path: dist/
+
+      - name: Publish package distributions to TestPyPI
```
Comment (miguelsousa) [inline .github/workflows/release.yml:65]: TestPyPI → PyPI ?

Comment (punchcutter) [reply .github/workflows/release.yml:65]: Ah, duh. I tested on test.pypi.org to make sure it was working correctly on my fork.

Comment (punchcutter) [reply .github/workflows/release.yml]: Removed "Test"

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,66 @@
+# This workflow will upload a Python Package using Twine when a release is created
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python#publishing-to-package-registries
```
Comment (miguelsousa) [inline .github/workflows/release.yml:2]: I think this is a stale comment; it's the GH action that uploads the artifacts, not twine.

Also, I think this is the current link https://docs.github.com/en/actions/use-cases-and-examples/building-and-testing/building-and-testing-python#publishing-to-pypi

Comment (punchcutter) [reply .github/workflows/release.yml:2]: I'm not even sure where that part came from. I was just grabbing a simple template and updating to make sure it worked correctly on a test repo.

**Review by miguelsousa — APPROVED** [review]

File: .github/workflows/release.yml (hunk 2 above)
Comment (punchcutter) [reply .github/workflows/release.yml]: Those comments aren't needed anyway so I removed them.

**Pull request merged**